src/main/scala/uk/gov/hmrc/mongo/ReactiveRepository.scala

```diff
@@ -79,6 +79,22 @@ abstract class ReactiveRepository[A <: Any, ID <: Any](collectionName: String,
       }
   }
 
+  override def bulkInsert(entities: Seq[A])(implicit ec: ExecutionContext): Future[MultiBulkWriteResult] = {
+    val docs = entities.map(toJsObject)
+    val failures = docs.collect { case Left(f) => f }
+    val successes = docs.collect { case Right(x) => x }
```

Comment:
If failures exist then no need to process the 'successes'. This could be moved into the else.